seqr/views/apis/igv_api_tests.py

```diff
@@ -37,27 +37,27 @@ def test_proxy_google_to_igv(self, mock_subprocess):
             'gsutil -u anvil-datastorage cat gs://fc-secure-project_A/sample_1.cram.gz',
             stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
 
+    @mock.patch('seqr.utils.file_utils.subprocess.Popen')
     @mock.patch('seqr.utils.file_utils.open')
-    def test_proxy_local_to_igv(self, mock_open):
-        mock_file = mock_open.return_value.__enter__.return_value
-        mock_file.__iter__.return_value = STREAMING_READS_CONTENT
-        mock_file.tell.side_effect = [0, 100, 200]
+    def test_proxy_local_to_igv(self, mock_open, mock_subprocess):
+        mock_subprocess.return_value.stdout = STREAMING_READS_CONTENT
+        mock_open.return_value.__enter__.return_value.__iter__.return_value = STREAMING_READS_CONTENT
 
         url = reverse(fetch_igv_track, args=[PROJECT_GUID, '/project_A/sample_1.bam.bai'])
         self.check_collaborator_login(url)
-        response = self.client.get(url, HTTP_RANGE='bytes=100-200')
+        response = self.client.get(url, HTTP_RANGE='bytes=100-250')
         self.assertEqual(response.status_code, 206)
-        self.assertListEqual([val for val in response.streaming_content], STREAMING_READS_CONTENT[:2])
-        mock_open.assert_called_with('/project_A/sample_1.bai', 'rb')
-        mock_file.seek.assert_called_with(100)
+        self.assertListEqual([val for val in response.streaming_content], STREAMING_READS_CONTENT)
+        mock_subprocess.assert_called_with(
+            'dd skip=100 count=150 bs=1 if=/project_A/sample_1.bai',
+            stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
```

Comment:
add an assertion for `mock_subprocess.stdout.__iter__` is called?

Reply:
I want this test to be consistent with the test above, which also calls subprocess. The mocked return value and checking `response.streaming_content` is already checking that it is being correctly iterated over so adding that line doesn't actually add anything